frontend/src/features/admin-form/settings/components/AuthSettingsSection/SecretKeyDownloadWhitelistFileModal.tsx

```diff
@@ -0,0 +1,282 @@
+    handleSubmit: verifyKeyPairAndDownloadWhitelistFile,
+    handleFileSelect: readValidateSetSecretKeyFormFieldFromFile,
+    handleOnClose: handleFormClose,
+    isSecretKeyUploaded,
+  }
+}
+
+export const SecretKeyDownloadWhitelistFileModal = ({
+  onClose,
+  isOpen,
+  publicKey,
+  downloadFileName,
+  formId,
+}: SecretKeyDownloadWhitelistFileModalProps): JSX.Element => {
+  const modalSize = useBreakpointValue({
+    base: 'mobile',
+    xs: 'mobile',
+    md: 'full',
+  })
+
+  const {
+    errors,
+    register,
+    fileUploadRef,
+    handleSubmit,
+    handleFileSelect,
+    handleOnClose,
+    isSecretKeyUploaded,
+  } = useSecretKeyWhitelistFileModal({
+    publicKey,
+    onClose,
+    formId,
+    downloadFileName,
+  })
+
+  return (
+    <Modal isOpen={isOpen} onClose={handleOnClose} size={modalSize}>
+      <ModalContent py={{ base: 'initial', md: '4.5rem' }}>
+        <ModalCloseButton />
+        {/* Hidden input field to trigger file selector, can be anywhere in the DOM */}
+        <Input
+          name="secretKeyFile"
+          ref={fileUploadRef}
+          type="file"
+          accept="text/plain"
+          onChange={handleFileSelect}
+          display="none"
+        />
+        <ModalHeader color="secondary.500">
+          <Container maxW="42.5rem">
+            Download whitelisted NRIC/FIN/UEN .csv file
+          </Container>
+        </ModalHeader>
+        <ModalBody whiteSpace="pre-wrap">
+          <Container maxW="42.5rem">
+            <FormActivationSvg mb="2rem" />
+            <form onSubmit={handleSubmit} noValidate>
+              <FormControl isRequired isInvalid={!!errors.secretKey} mb="1rem">
+                <FormLabel>Enter or upload Secret Key</FormLabel>
+                <Stack direction="row" spacing="0.5rem">
+                  <Input
+                    type="password"
+                    {...register(SECRET_KEY_NAME, {
+                      required: "Please enter the form's secret key",
+                      pattern: {
+                        value: SECRET_KEY_REGEX,
+                        message: 'The secret key provided is invalid',
+                      },
+                    })}
+                    placeholder="Enter or upload your Secret Key to continue"
+                  />
+                  <IconButton
+                    variant="outline"
+                    aria-label="Pass secret key from file"
+                    icon={<BiUpload />}
+                    onClick={() => fileUploadRef.current?.click()}
+                  />
+                </Stack>
+                <FormErrorMessage>{errors.secretKey?.message}</FormErrorMessage>
+              </FormControl>
+              <Button
+                rightIcon={<BiRightArrowAlt fontSize="1.5rem" />}
+                type="submit"
+                isFullWidth
+                isDisabled={!isSecretKeyUploaded}
+              >
+                Download
+              </Button>
+            </form>
```

Comment:
Can this reuse the component from `frontend/src/components/SecretKeyVerificationInput/SecretKeyVerificationInput.tsx`?

**Context**
We have additional changes that updates this Input field to support drag n drop.

Reply:
Resolved by setting common `SecretKeyFormModal` component which uses the reusable `useSecretKeyForm` hook